### PR TITLE
Fixes #13604 - Use content_type options in the api

### PR DESCRIPTION
**Pull request by daviddavis**
Waiting on #5755 so I can also update the api param descriptions.

### CODE REVIEW

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@ class Api::V2::RepositoriesController < Api::V2::ApiController
     param :erratum_id, String, :desc => N_("Id of an erratum to find repositories that contain the erratum")
     param :rpm_id, String, :desc => N_("Id of a package to find repositories that contain the rpm")
     param :library, :bool, :desc => N_("show repositories in Library and the default content view")
-    param :content_type, String, :desc => N_("limit to only repositories of this time")
+    param :content_type, RepositoryTypeManager.repository_types, :desc => N_("limit to only repositories of this time")
```
Comment (daviddavis) [inline app/controllers/katello/api/v2/repositories_controller.rb]: `limit to only repositories of this type (either 'yum', 'puppet', 'ostree', or 'docker')`

**Comment by daviddavis** [comment]
[test]

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ class Api::V2::RepositoriesController < Api::V2::ApiController
       param :url, String, :desc => N_("repository source url")
       param :gpg_key_id, :number, :desc => N_("id of the gpg key that will be assigned to the new repository")
       param :unprotected, :bool, :desc => N_("true if this repository can be published via HTTP")
-      param :content_type, String, :required => true, :desc => N_("type of repo (either 'yum', 'puppet', 'docker', or 'ostree')")
+      param :content_type, RepositoryTypeManager.creatable_repository_types.keys, :required => true, :desc => N_("type of repo (either 'yum', 'puppet', 'docker', or 'ostree')")
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/repositories_controller.rb]: Shouldn't this be part of the description rather than the type?

Comment (daviddavis) [inline app/controllers/katello/api/v2/repositories_controller.rb]: Nah, I should probably actually remove it from the description. If you use an enum as the type, you get the proper doc/validation:

screen shot 2016-02-23 at 9 05 08 am

**Comment by jlsherrill** [comment]
the more you know

ACK

**Pull request merged**